schema-lending-extended.graphql

```diff
@@ -0,0 +1,1173 @@
+  hourlyRepayUSD: BigDecimal!
+
+  ##### Token Balances #####
+
+  " Amount of input token in the market. "
+  inputTokenBalance: BigInt!
+
+  " Price per share of input token in USD "
+  inputTokenPriceUSD: BigDecimal!
+
+  " Total supply of output token "
+  outputTokenSupply: BigInt!
+
+  " Price per share of output token in USD "
+  outputTokenPriceUSD: BigDecimal!
+
+  " Amount of input token per full share of output token. Only applies when the output token exists "
+  exchangeRate: BigDecimal
+
+  " Per-block reward token emission as of the current block normalized to a day (not hour), in token's native amount. This should be ideally calculated as the theoretical rate instead of the realized amount. "
+  rewardTokenEmissionsAmount: [BigInt!]
+
+  " Per-block reward token emission as of the current block normalized to a day (not hour), in USD value. This should be ideally calculated as the theoretical rate instead of the realized amount. "
+  rewardTokenEmissionsUSD: [BigDecimal!]
+}
+
+##############################
+##### Account-Level Data #####
+##############################
+
+type Account @entity {
+  " { Account address } "
+  id: ID!
+  
+  " Number of open positions this account has "
+  positionCount: Int!
+
+  " All open positions that belong to this account "
+  positions: [Position!]!
+
+  " Number of deposits this account made "
+  depositCount: Int!
+
+  " All deposit events of this account "
+  deposits: [Deposit!]! @derivedFrom(field: "from")
+
+  " Number of withdrawals this account made "
+  withdrawCount: Int!
+
+  " All withdraw events of this account "
+  withdraws: [Withdraw!]! @derivedFrom(field: "to")
+
+  " Number of borrows this account made "
+  borrowCount: Int!
+
+  " All borrow events of this account "
+  borrows: [Borrow!]! @derivedFrom(field: "to")
+
+  " Number of repays this account made "
+  repayCount: Int!
+
+  " All repay events of this account "
+  repays: [Repay!]! @derivedFrom(field: "from")
+
+  " Number of times this account liquidated a position "
+  liquidateCount: Int!
+
+  " All liquidation events where this account was the liquidator "
+  liquidates: [Liquidate!]! @derivedFrom(field: "from")
+
+  " Number of times this account has been liquidated "
+  liquidationCount: Int!
+
+  " All liquidation events where this account got liquidated "
+  liquidations: [Liquidate!]! @derivedFrom(field: "liquidatee")
+}
+
+type Position @entity {
+  " { Account address }-{ Market address }-{ Counter } "
+  id: ID!
+
+  " Account that owns this position "
+  account: Account!
+
+  " The market in which this position was opened "
+  market: Market!
+
+  " The hash of the transaction that opened this position "
+  hash: String!
+
+  " Side of the position (either lender or borrower) "
+  side: PositionSide!
+
+  " Whether this position has been closed or not "
+  closed: Boolean!
+
+  " The token in the position "
+  token: Token!
+
+  " Token balance in this position, in native amounts "
+  balance: BigInt!
```

Comment:
How are we handling this changing over time?

Reply:
It may make the schema overly circular/bloated, but should we hold the events of a position with this entity?

Reply:
`PositionSnapshot` has `balance` change over time

Reply:
Yeah, I've thought about that. What are the use cases you have in mind?

Reply:
The first thing that comes to mind would be aggressively tracking high importance positions under stress. Lot's of people rn are watching the celsius positions continue to top up on collateral, this may be the easiest way to go about that. It is a niche use case though.